apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/ITraceContext.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright 2021 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// A trace context.
+    /// </summary>
+    public interface ITraceContext
+    {
+        /// <summary>The trace id or null if none is available.</summary>
+        /// <remarks>
+        /// This corresponds to the trace_id field of the Trace message in the API.
+        /// See https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v1#trace
+        /// for more information.
+        /// </remarks>
+        string TraceId { get; }
+
+        /// <summary>The span id or null if none is available.</summary>
+        /// <remarks>
+        /// This corresponds to the span_id field of the TraceSpan message in the API.
+        /// See https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v1#tracespan
+        /// for more information.
+        /// </remarks>
+        ulong? SpanId { get; }
+
+        /// <summary>
+        /// Whether this request should be traced or not.
+        /// May be null, which means that there's not enough information to know
+        /// wether this request should be traced or not.
+        /// </summary>
+        /// <remarks>
+        /// This is also sometimes understood as information sent by the requester about
+        /// whether it recorded trace information for this request or not, i.e. whether
+        /// this request was sampled on the requester side or not.
+        /// </remarks>
+        bool? ShouldTrace { get; }
```

Comment:
Should we add a flag dictionary instead, to support possible future flags? See https://w3c.github.io/trace-context/#trace-flags

Reply:
Ah yes, missed that earlier. Not sure. If there are multiple flags, would we have the concept of *some* of them not being known, or is it just "there are flags" or "there aren't flags"?
I'm tempted not to try to do too much here, especially if we're expecting OpenTelemetry to take over in the long term.

Reply:
I was thinking of an enum with the name of the flags we accept, right now it would be just `Flags.Sampled`. And a dictionary FlagsEnum -> bool. If we care about a new flag we add it to the enum, else we don't add it to the enum. Implementations of ITraceContext are free to add more flags or in general more information that we don't use but user code does.

But I agree that it's better to keep things simples given the plans for the future.